incubator/virtualcluster/pkg/syncer/resources/pod/dws.go

```diff
@@ -174,7 +176,35 @@ func (c *controller) reconcilePodCreate(cluster, namespace, name string, vPod *v
 		return fmt.Errorf("failed to find nameserver: %v", err)
 	}
 
-	err = conversion.VC(c.multiClusterPodController, cluster).Pod(pPod).Mutate(vPod, vSecret, pSecret, services, nameServer)
+	var ms = []conversion.PodMutator{
+		conversion.PodMutateDefault(vPod, vSecret, pSecret, services, nameServer),
+		conversion.PodMutateAutoMountServiceAccountToken(c.config.DisableServiceAccountToken),
+		conversion.PodAddExtensionMeta(vPod),
+	}
+
+	if c.config.EnableTenantKubeConfig {
+		clientConfig, err := c.multiClusterPodController.GetClusterClientConfig(cluster)
+		if err != nil {
+			return fmt.Errorf("failed to get cluster config")
+		}
+
+		kubeConfigBytes, err := createKubeConfigByServiceAccount(tenantClient.CoreV1().ServiceAccounts(vPod.Namespace), tenantClient.CoreV1().Secrets(vPod.Namespace), clientConfig, saName)
+		if err != nil {
+			return fmt.Errorf("failed to create kubeconfig from service account: %v", err)
+		}
+
+		secret := conversion.BuildKubeConfigSecret(cluster, vPod, kubeConfigBytes)
+		secret.Namespace = targetNamespace
+
```

Comment:
First I want to add directly `ownerReference` to the secret object. But We should first create the Pod, then we got the UID. There is a deadlock.